core-bundle/assets/controllers/sticky-observer-controller.js

```diff
@@ -0,0 +1,25 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    connect() {
+        this.observer = new IntersectionObserver(this.#handleIntersect.bind(this), this.observerOptions);
+        this.observer.observe(this.element);
+    }
+
+    disconnect() {
+        this.observer.disconnect();
+    }
+
+    get observerOptions() {
+        return {
+            threshold: 1.0,
+            rootMargin: '0px 0px -1px 0px',
+        };
+    }
+
+    #handleIntersect(entries) {
+        for (const entry of entries) {
+            this.element.classList.toggle('is-stuck', !entry.isIntersecting);
```

Comment:
```suggestion
            this.element.classList.toggle('is--stuck', !entry.isIntersecting);
```

I'd rather have it as a state, similar to the `is--hidden` in the operations menu controller

Reply:
But `is--stuck` would be a modifier `--stuck` on the element with the CSS class `is` according to BEM. That is not correct (and neither is `is--hidden` then).